pkg/operator/resource_reconciler.go

```diff
@@ -457,3 +489,16 @@ func ListMatchingNamespaces(selector labels.Selector, nsInf cache.SharedIndexInf
 	}
 	return ns, nil
 }
+
+// hasOwnership returns true if the controller is the "owner" of the object.
+// Whether it's owner is determined by the value of 'controllerID'
+// annotation. If the value matches the controllerID then it owns it, or if the
+// controllerID is "" and there's no annotation set.
+func (rr *ResourceReconciler) hasOwnership(obj metav1.Object) bool {
+	if owner, ok := obj.GetAnnotations()[controllerIDLabel]; ok {
+		return owner == rr.controllerID
+	}
+
+	level.Debug(rr.logger).Log("msg", "%s object has been created by another prometheus operator, controllerID does not match, please be careful deploying multiple operators", rr.resourceKind)
```

Comment:
the code needs to be rearranged to always log the debug message (currently it only logs if the annotation is missing).
```suggestion
	level.Debug(rr.logger).Log("msg", "skipping object not managed by the controller", "object", fmt.Sprintf("%s/%s", o.GetNamespace(), o.GetName()), "object_controller_id", obj.GetAnnotations()[controllerIDLabel], "controller_id", rr.controllerID)
```